dashboard/app/controllers/concerns/azure_text_to_speech.rb

```diff
@@ -0,0 +1,82 @@
+require 'cdo/honeybadger'
+require 'cdo/languages'
+require 'net/http'
+require 'dynamic_config/gatekeeper'
+
+module AzureTextToSpeech
+  # Azure authentication token is valid for 10 minutes, so cache it for 9.
+  # https://docs.microsoft.com/en-us/azure/cognitive-services/speech-service/rest-text-to-speech#authentication
+  TOKEN_CACHE_TTL = 9.minutes.freeze
+
+  # Requests an authentication token from Azure, or returns the cached token if still valid.
+  def self.get_token
+    return nil unless allowed?
+
+    Rails.cache.fetch("azure_speech_service/token", expires_in: TOKEN_CACHE_TTL) do
```

Comment:
this is the only functional change here -- everything else is moving code/tests from LevelsHelper to AzureTextToSpeech